main.tf

```
@@ -29,7 +29,7 @@ locals {
 }
 
 resource "null_resource" "global_secondary_index_names" {
-  count = "${length(var.global_secondary_index_map)}"
+  count = "${(var.enabled ? 1 : 0 ) * length(var.global_secondary_index_map)}"
```

```suggestion
  count = "${(var.enabled == "true" ? 1 : 0 ) * length(var.global_secondary_index_map)}"
```

Let's use explicit typing since `var.enabled` is a string var (as recommended by TF to use strings instead of bool in variables).
We use `var.enabled == "true"` in all modules, or we create a local 
```
enabled = "${var.enabled == "true" ? true : false}"
```

your choice, thanks

---

By using only `var.enabled` it will both evaluate "true" and true as true. But happy to make adjustments.